Assets/New Babylon/Scripts/AudioArchitecture/AudioHandler.cs

```diff
@@ -55,5 +55,42 @@ public virtual void RandomizePitch(float min, float max)
         float randomPitch = Random.Range(min, max);
         myAudioSource.pitch = randomPitch;
     }
+
+    [Header("Multi voice set up")]
+    public int voiceCounter;
+    public AudioSource[] voices;
+    //called to play sounds 
+    public void PlaySoundMultipleAudioSources(AudioClip[] soundArray)
+    {
+        int randomSound = Random.Range(0, soundArray.Length);
+
+        voiceCounter = CountUpArray(voiceCounter, voices.Length - 1);
+
+        voices[voiceCounter].PlayOneShot(soundArray[randomSound]);
+    }
+
+    public int currentSound = 0;
+    //counts up through sound array, in order 
+    public void PlaySoundUp(AudioClip[] soundArray)
+    {
+        currentSound = CountUpArray(currentSound, soundArray.Length - 1);
```

Comment:
There is a better way to do this. 

I can't tell if you are trying to sum the total array or just have a list or something but there is a better way. Look into a reduce function.